android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java

```diff
@@ -1093,7 +1093,15 @@ private void dispatchCallback(String name, KrollDict data) {
 
 		data.put("source", activityProxy);
 
-		activityProxy.callPropertyAsync(name, new Object[] { data });
+		// TIMOB-19903
+		if (TiApplication.getInstance().runOnMainThread()) {
+			// We must call this synchornously to ensure it happens before we release the Activity reference on the V8/Native side!
+			activityProxy.callPropertySync(name, new Object[] { data });
```

Comment:
Aren't we only releasing the references in onDestroy()? Making it sync on every on* methods seem a bit excessive.